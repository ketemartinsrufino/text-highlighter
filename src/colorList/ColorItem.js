import { useState } from 'react';
import styled from 'styled-components';

const ColorItemStyled = styled.div`
    border: ${({isSelected}) => isSelected ? 2: 1}px black solid;
    height:30px;
    width: 50px;
    background-color: ${({color}) => color};

    :hover {
        cursor: pointer:
    }
`;

const ColorItem = ({color}) => {

    return <ColorItemStyled 
            color="red" 
            isSelected={this.isSelected("red")}
            onClick={this.getOnClick("red")}
        />
}