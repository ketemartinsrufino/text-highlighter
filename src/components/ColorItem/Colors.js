export const COLORS = {
  red: '#ef5350', 
  yellow: '#ffeb3b', 
  green: '#8eda8e'
}