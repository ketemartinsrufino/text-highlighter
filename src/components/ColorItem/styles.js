import styled from 'styled-components';

export const ColorItemStyled = styled.div`
  background-color: ${({color}) => color};
  border-radius: 50%;
  height: 30px;
  width: 30px;
  margin: 0 5px;
  border: 1px #a97c7c solid;

  ${({ isSelected }) => isSelected && `
    border: 2px black solid
  `}

  :hover {
    cursor: pointer:
  }
`;