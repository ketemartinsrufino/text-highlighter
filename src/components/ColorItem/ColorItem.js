import React from 'react';
import PropTypes from 'prop-types';
import { ColorItemStyled } from './styles';

const ColorItem = ({ color, isSelected, onClick }) => (
  <ColorItemStyled
    color={color}
    isSelected={isSelected}
    onClick={() => onClick(color)}
    className={`ColorItem${isSelected ? ' selected' : ''}`}
  />
)

ColorItem.defaultProps = {
  isSelected: false
}

ColorItem.propTypes = {
  isSelected: PropTypes.bool,
  color: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
}

export default ColorItem;