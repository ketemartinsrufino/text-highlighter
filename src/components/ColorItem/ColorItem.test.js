import React from 'react';
import { shallow } from 'enzyme';
import ColorItem from './ColorItem';
import toJson from 'enzyme-to-json';

describe('ColorItem', () => {
  it('if receive "red" color, will render a "red" component', () => {
    const component = shallow(<ColorItem color="red" onClick={jest.fn()} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('if receive "isSelected" prop, will render a bigger border', () => {
    const component = shallow(<ColorItem color="red" isSelected={true} onClick={jest.fn()} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});