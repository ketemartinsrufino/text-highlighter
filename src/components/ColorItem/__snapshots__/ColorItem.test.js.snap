// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorItem if receive "isSelected" prop, will render a bigger border 1`] = `
<styled.div
  className="ColorItem selected"
  color="red"
  isSelected={true}
  onClick={[Function]}
/>
`;

exports[`ColorItem if receive "red" color, will render a "red" component 1`] = `
<styled.div
  className="ColorItem"
  color="red"
  isSelected={false}
  onClick={[Function]}
/>
`;
